src/metabase/public_settings.clj

```diff
@@ -201,6 +201,10 @@
     (application-name-for-setting-descriptions))
   :default    "en"
   :visibility :public
+  :getter     (fn []
+                (let [value (setting/get-value-of-type :string :site-locale)]
+                  (when (i18n/available-locale? value)
+                    value)))
```

Comment:
Not strictly needed for this fix but I noticed the Setting getter did not validate its values and happily returned invalid ones; I fixed this